package.json

```diff
@@ -187,35 +190,35 @@
   },
   "private": true,
   "scripts": {
-    "api.update": "tsm scripts/index.ts --tsc --api --dev",
-    "build": "tsm scripts/index.ts",
+    "api.update": "node --loader ts-node/esm --require ./scripts/runBefore.ts scripts/index.ts --tsc --api --dev",
```

Comment:
The `--loader` flag is [discouraged](https://nodejs.org/api/cli.html#--experimental-loadermodule) in favor of `--import`, are you sure this is the right one? Also ts-node does not work with the --import flag. See: https://github.com/TypeStrong/ts-node/issues/1909

I suggest to consider these other options instead:

1. [tsimp](https://www.npmjs.com/package/tsimp) - uses tsc + caching;
2. [tsx](https://www.npmjs.com/package/tsx) - uses esbuild;
3. Anything Node.js itself has to offer: [`--experimental-strip-types`](https://nodejs.org/api/cli.html#--experimental-strip-types) or [`--experimental-transform-types`](https://nodejs.org/api/cli.html#--experimental-transform-types) flag, but they are available since Node v22 end they are experimental;

Reply:
@JerryWu1234 Did you see this? I think tsimp is the best fit in our case, because it's only for build tooling, is light-weight and it will auto-start a daemon that caches.

Because we need to preload some globals, we still need the same approach as tsm.cjs, but it would instead be e.g. `scripts/tsimp.cjs`, which would `import "tsimp"` and then splice() `process.argv[1]` (I think) and then `await import(process.argv[1])` which would run the script as if node ran it directly.

Reply:
Joining in on the fun :steam_locomotive: 

`tsimp` doesn't work with [Typescript v5.6.2](https://github.com/tapjs/tsimp/issues/29) and the maintainer stated that [the project is staled and isn't stable](https://github.com/tapjs/tsimp/issues/29#issuecomment-2397517830)

So back to square one!